### PR TITLE
Separated into files

### PR DESCRIPTION
**Pull request by josh-flori**

### CODE REVIEW

File: School.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+void School::terminate(Teacher teacher)
+{
+    teacher.employmentStatus = 0;
+}
+void School::giveBonus(Teacher teacher, float amount)
+{
+    teacher.spotBonus = amount;
+}
```
Comment (matkatmusic) [inline School.cpp:16]: These `Teacher` instances are being passed by copy.
This code does not modify the instance that was passed to the function.
it modifies a separate instance entirely.
This is a common issue with new programmers or programmers who are coming from garbage-collected languages (python, C#). 
**Pass by reference instead.**
You'll need to revise your declarations of these functions
- [ ] revise

File: Conglomerate.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    }   
+}
+void compareFranchises(Franchise firstFranchise, Franchise secondFranchise)
+{
+    if(firstFranchise.assets > secondFranchise.assets)
+    {
+        std::cout << "First franchise has more assets\n\n";
+    }
+    std::cout << "Second franchise has more assets\n\n";
+}
```
Comment (matkatmusic) [inline Conglomerate.cpp:23]: These `Franchise` instances are being passed by copy.
This code does not modify the instance that was passed to the function.
it modifies a separate instance entirely.
This is a common issue with new programmers or programmers who are coming from garbage-collected languages (python, C#). 
**Pass by reference instead.**
You'll need to revise your declarations of these functions
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: Franchise.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Franchise::~Franchise() 
+{
+    std::cout << "\nFranchise destructing\n";
+}
+float Franchise::getEstimatedRevenue(float averageLocationRevenue)
+{
+    float estRev = 0;
+    for(int i = 0; i < locations; i++)
+    {
+        estRev += averageLocationRevenue;
```
[the remark this answers is not in the document]
Comment (josh-flori) [reply Franchise.cpp:14]: I am familiar with this idea and know not to use it for indexing, but for FOR loops specifically it doesn't make a difference does it? 
Every for loop I see online uses i++ and testing yields no difference in what value i holds in the body of the loop.

File: Student.cpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    gpa = finalGPA;
+    for(int i = 1; i <= coursesPassed; i ++)
+    {
+        std::cout << "Successfully Completed Course" + std::to_string(i) + "\n\n";
+    }
+}
+void Student::withdraw(int courseHours)
+{
+    enrolledCourses -= 1;
+    totalHours -= courseHours;
```
[the remark this answers is not in the document]
Comment (josh-flori) [reply Student.cpp:30]: see comment in chat. I don't understand the issue or what form the correct approach takes

File: Teacher.cpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    }
+}
+void Teacher::transferStudents(int studentCount)
+{
+    totalStudents -= studentCount;
+}
+void Teacher::completeYear()
+{
+    yearsTillRetirement -= 1;
+    totalStudents = 0;
```
[the remark this answers is not in the document]
Comment (josh-flori) [reply Teacher.cpp:23]: see comment in chat. I don't understand the issue or what form the correct approach takes

File: Conglomerate.h [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+#include "LeakedObjectDetector.h"
+
+#include "Franchise.h"
+
+struct Conglomerate
+{
+    ~Conglomerate();
+    void cookTheBooks(Franchise& franchise, float amount);
```
Comment (matkatmusic) [inline Conglomerate.h:9]: Where is your constructor?
This class needs a constructor, as does every other class.
Project 5 Part 1 required you to write constructors
Declare a constructor.
Implement it in the associated cpp file.
- [ ] revise

File: Location.h [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    void makeSale(float price, float expenses);
+    void closeLocation();
+    void hireEmployee(int numberHired = 1);
+    
+    float locationRating = 4.3f;
+    float revenue = 55000.3f;
+    float profit = 5000.2f;
+    int employeeCount = 1;
+    int openPositionCount = 0;
+};
```
Comment (matkatmusic) [inline Location.h:17]: Delete this class.  It is a nested class in `Franchise`
- [ ] revise

File: Franchise.cpp [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    return estRev;
+}
+void Franchise::launchNewLocation(float debt, float capitalRequirements)
+{
+    liabilities += debt;
+    assets -= capitalRequirements;
+    locations += 1;
+}
```
Comment (matkatmusic) [inline Franchise.cpp:23]: use `++locations;`
- [ ] revise

File: Franchise.cpp [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+void Franchise::setRating(float newRating)
+{
+    ratingCount += 1;
```
Comment (matkatmusic) [inline Franchise.cpp:26]: use `++ratingCount;`
- [ ] revise

File: Student.cpp [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    std::cout << "Instantiating student\n";
+}
+Student::~Student()  
+{
+    std::cout << "Student destructing\n";
+}
+
+void Student::enroll(int courseHours)
+{
+    enrolledCourses += 1;
```
Comment (matkatmusic) [inline Student.cpp:16]: use `++enrolledCourses;`
- [ ] revise

File: Student.cpp [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+{
+    gpa = finalGPA;
+    for(int i = 1; i <= coursesPassed; ++i)
+    {
+        std::cout << "Successfully Completed Course" + std::to_string(i) + "\n\n";
+    }
+}
+void Student::withdraw(int courseHours)
+{
+    enrolledCourses -= 1;
```
Comment (matkatmusic) [inline Student.cpp:29]: use `--i;` syntax
- [ ] revise

File: Teacher.cpp [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        totalCourseHours += 3;
+    }
+}
+void Teacher::transferStudents(int studentCount)
+{
+    totalStudents -= studentCount;
+}
+void Teacher::completeYear()
+{
+    yearsTillRetirement -= 1;
```
Comment (matkatmusic) [inline Teacher.cpp:26]: use `--i;` syntax
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**